DragAndDrop/ViewController.swift

```diff
@@ -8,11 +8,16 @@
 
 import UIKit
 
-class ViewController: UIViewController {
+class ViewController: UIViewController, ViewDelegate {
 
+    var mainView:View!
```

Comment:
More :worried: 

This one is a bit trickier to get rid of...
We can't just initialize this one here, since it depends on `self.frame`
And self is not guaranteed to be valid at init time. 
We could make this a full blown optional, I would vote that as an improvement. 

My suggestion is to use lazy evaluation. We can put the lazy keyword in front of this and then instantiate it with a closure. The closure is then run the first time the value is accessed. 
```swift
lazy var mainView: View = {
    let v = View(frame: self.view.frame)
    v.delegate = self
    self.view.addSubView(v)
    return v
}()
```